### ISSUE TITLE
@JsonAnySetter with @JsonUnwrapped: deserialization fails with arrays

### ISSUE THREAD

**Issue by hdave** [FasterXML/jackson-databind#349]
My application makes heavy use of the @JsonAnySetter/Getter annotations which is currently throwing an exception when deserializing an JSON array of strings as a property value:

```
Can not deserialize instance of java.lang.Object out of END_ARRAY token at [Source: org.apache.cxf.transport.http.AbstractHTTPDestination$1@7a5d231e; line: 22, column: 5] (through reference chain: com.mycorp.core.web.rest.dto.ItemDTO["ZoomLinks"])
at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:164)
at com.fasterxml.jackson.databind.DeserializationContext.mappingException(DeserializationContext.java:575)
at com.fasterxml.jackson.databind.DeserializationContext.mappingException(DeserializationContext.java:569)
at com.fasterxml.jackson.databind.deser.std.UntypedObjectDeserializer.deserialize(UntypedObjectDeserializer.java:90)
at com.fasterxml.jackson.databind.deser.SettableAnyProperty.deserialize(SettableAnyProperty.java:98)
at com.fasterxml.jackson.databind.deser.SettableAnyProperty.deserializeAndSet(SettableAnyProperty.java:88)
at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeWithUnwrapped(BeanDeserializer.java:515)
at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:271)
at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:121)
at com.fasterxml.jackson.databind.ObjectReader._bind(ObjectReader.java:1179)
at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:635)
```

Here is the Java code:

``` @JsonAnySetter
public void addProperty(String key, Object value) {
if (props == null)
{ props = new HashMap<String, Object>(); }
props.put(key, value);
}
@JsonAnyGetter
public Map<String, Object> getProperties()
{ return props; }
```

and here is some sample JSON that will fail:

```
{ "_type" : "IST",
 "_spacename" : "Foo Models",
 "_name" : "BLAH-New",
 "_description" : "namespace.name: X THIN FIR.DR-WD12-New",
 "ZoomLinks": [ "foofoofoofoo", "barbarbarbar" ] }
```

**Comment by borball** [comment]
It seems JsonAnySetter and JsonUnwrapped can work together in 2.9.0-SNAPSHOT but not in 2.8.6.  I have added one test case here:
https://github.com/borball/jackson-test/blob/master/src/test/java/com/fasterxml/jackson/test/JacksonTest.java#L15-L19
With 2.9.0-SNAPSHOT I can successfully get bean object refundQuery but with 2.8.6, it is null.

**Comment by atramos** [comment]
This seems broken in 2.9.9, but in a different way... the `@JsonUnwrapped` fields are duplicated in the `@JsonAnySetter`. Is this a regression or always been this way?

**Comment by cowtowncoder** [comment]
I think that the way things are implemented duplicates would always have occurred.

Later issue #1811 cover the problem: I hope this can be resolved in 3.0.
